abnf/odata-aggregation-abnf.txt

```diff
@@ -124,7 +123,8 @@ aggregateMethod = %s"sum"
                 / nonprimAggMethod
 nonprimAggMethod = %s"countdistinct"
                  / namespace "." odataIdentifier ; custom aggregation methods may work on non-primitive values
-aggregateCount  = ( [ aggrCastPath "/" ] aggrPropPath / aggrCastPath ) %s"/$count"
+aggregateCount  = %s"$count"
+                / ( [ aggrCastPath "/" ] aggrPropPath / aggrCastPath ) %s"/$count"

```

Comment:
Can we clean up aggrPropPath, aggrPropStep, and aggrCastPath?

With aggrCastPath we have given up the principle of "complex type cast only after complex property and entity type cast only after navigation property", so there is potential to shorten aggrPropStep, and maybe collapse some of the numerous routes through our ABNF rules.